src/spaces.rs

```diff
@@ -176,22 +176,24 @@ fn compute_halstead_and_mi<'a, T: ParserTrait>(state: &mut State<'a>) {
 
 fn finalize<'a, T: ParserTrait>(state_stack: &mut Vec<State<'a>>, diff_level: usize) {
     for _ in 0..diff_level {
-        if state_stack.len() <= 1 {
+        if state_stack.len() < 1 {
+            break;
+        } else if state_stack.len() == 1 {
```

Comment:
the first line of this block and the next one are almost the same

Reply:
I can't move that `else if` outside the loop though. What is the problem here?

Reply:
I thought you did almost the same thing in the two branches but I think your code is correct because we need to keep the last value int he stack.

Reply:
Oh, I see. I thought there was a problem with the loop, sorry :)